sdk/workloadssapvirtualinstance/Azure.ResourceManager.WorkloadsSapVirtualInstance/src/Generated/Models/SapVirtualMachineConfiguration.cs

```diff
@@ -0,0 +1,89 @@
+        /// <list type="bullet">
+        /// <item>
+        /// <term>BinaryData.FromObjectAsJson("foo")</term>
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("\"foo\"")</term>
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromObjectAsJson(new { key = "value" })</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("{\"key\": \"value\"}")</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// </list>
+        /// </para>
+        /// </summary>
+        private IDictionary<string, BinaryData> _serializedAdditionalRawData;
+
+        /// <summary> Initializes a new instance of <see cref="SapVirtualMachineConfiguration"/>. </summary>
+        /// <param name="vmSize"> The virtual machine size. </param>
+        /// <param name="imageReference"> The image reference. </param>
+        /// <param name="osProfile"> The OS profile. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="vmSize"/>, <paramref name="imageReference"/> or <paramref name="osProfile"/> is null. </exception>
+        public SapVirtualMachineConfiguration(string vmSize, SapImageReference imageReference, SapOSProfile osProfile)
+        {
+            Argument.AssertNotNull(vmSize, nameof(vmSize));
+            Argument.AssertNotNull(imageReference, nameof(imageReference));
+            Argument.AssertNotNull(osProfile, nameof(osProfile));
+
+            VmSize = vmSize;
+            ImageReference = imageReference;
+            OSProfile = osProfile;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="SapVirtualMachineConfiguration"/>. </summary>
+        /// <param name="vmSize"> The virtual machine size. </param>
+        /// <param name="imageReference"> The image reference. </param>
+        /// <param name="osProfile"> The OS profile. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal SapVirtualMachineConfiguration(string vmSize, SapImageReference imageReference, SapOSProfile osProfile, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            VmSize = vmSize;
+            ImageReference = imageReference;
+            OSProfile = osProfile;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="SapVirtualMachineConfiguration"/> for deserialization. </summary>
+        internal SapVirtualMachineConfiguration()
+        {
+        }
+
+        /// <summary> The virtual machine size. </summary>
+        public string VmSize { get; set; }
```

Comment:
should this be `long VmSizeInGB`?

Reply:
Not disk size, but scale size like 'Standard_E4ds_v4'